source/mobile/mobile-hpns.rst

```diff
@@ -35,10 +73,43 @@ Follow these steps to set up HPNS:
 
 4. Review the Mattermost Terms of Service and the Mattermost Privacy Policy, then select the box "I understand and accept the Mattermost Hosted Push Notification Service Terms of Service and Privacy Policy" to acknowledge that you understand the terms of use.
 
-.. Note:: 
+.. note:: 
 
-  The default **Push Notification Server** address is ``https://push.mattermost.com`` and the server is hosted inside the United States. Mattermost also offers a push notification server hosted in Germany. If you wish to use that server, please update the **Push Notification Server** address to ``https://hpns-de.mattermost.com/``.
+  The default **Push Notification Server** address is ``https://push.mattermost.com``. The server is hosted inside the United States. Mattermost also offers a push notification server hosted in Germany. If you wish to use the server in Germany, update the **Push Notification Server** address to ``https://hpns-de.mattermost.com/``.
 
 5. Select **Save**
 
-After setup, :doc:`test push notifications <mobile-testing-notifications>` to confirm they are working.
+After setup, test push notifications to confirm they are working.
+
+Mattermost Push Notification Service (MPNS)
+-------------------------------------------
+
+Mattermost provides a self-hosted push proxy you can deploy called the `Mattermost Push Notification Service (MPNS) <https://docs.mattermost.com/deployment/deployment.html#push-notification-service>`__. 
+
+Enabling MPNS in Mattermost
```

Comment:
should we add a note that this is not available in Cloud?

Reply:
Yes! Thanks!